auth/aws_test.go

```diff
@@ -0,0 +1,49 @@
+package auth
+
+import (
+	"io"
+	"net/http"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+)
+
+func TestGetSTSHeaderAndBody(t *testing.T) {
+
+	testCases := []struct {
+		name              string
+		getCallerIdentity func() *request.Request
+		expectedHeader    string
+		expectedBody      string
+		expectedError     error
+	}{
+		{"good Response", func() *request.Request {
```

Comment:
question: Are there other tests we need other than the happy path?

Reply:
Good Catch! We will fix this next GCP and AZUR Pr